### DISCUSSION TITLE
建议新增负载均衡功能

### DISCUSSION THREAD

**Discussion by sophauer** [zfl9/ss-tproxy#251] [Ideas]
如标题,如果能有负载均衡或是一个节点失效后.,自动切换到另一个节点的功能就太好了

**Comment by zfl9** [comment]
想法很好，但是这个功能说实话，并不归 ss-tproxy 自身。这个其实是 本机代理进程 实现的哈。

节点负载均衡，这个可以看下 clash，readme 有介绍。不过 clash 貌似不支持 xray 的一些协议。

**Comment by sophauer** [comment]
我看merlin上的 https://github.com/hq450/fancyss
是通过脚本实现的
提供给你参考

**Comment by zfl9** [comment]
嗯，不过我感觉可以使用 haproxy 的 TCP 代理来进行负载均衡，后面研究下。

**Comment by sophauer** [comment]
期待~

**Comment by ir1ka** [comment]
负载均衡不是可以通过代理进程 reuse port 来实现吗？多启动几个代理进程，监听相同端口（reuse port）即可。

**Reply by ir1ka** [reply]
某个节点失效，相应的代理进程退出或不再监听端口即可。

**Reply by sophauer** [reply]
你说的这个叫故障转移,不是负载均衡

**Reply by ir1ka** [reply]
reuse port 是 linux kernel 帮你做负载均衡。同时当然也可以做故障转移，比如故障节点的代理进程退出即可。

**Reply by ir1ka** [reply]
参考链接：https://lwn.net/Articles/542629/

**Reply by sophauer** [reply]
额,那么,大佬能不能麻烦你做一个配置示范.解说一下,实在不晓得该怎么实现?比如我同时启用两个SS连接,本地端口port reuse都在60080,那我该怎么让它在访问不了google.com时,切换节点

**Reply by ir1ka** [reply]
代理进程理应能够知道自己的的连接是否仍然有效，不再有效则应重启或退出。
如果代理进程无法知道，则可以先用监听普通端口，然后用一个 ipt2socks 转换为透明代理并启用 reuse port。用此方法启动两个或多个代理，另起一个进程监控代理进程监听的普通端口是否有效，当不再有效时 kill 它对应的 ipt2socks 进程，当监控到有效时再拉起该 ipt2socks 即可。

**Reply by ir1ka** [reply]
这样一个即可实现**负载均衡**又能实现**故障转移**的简单方案。

**Reply by ir1ka** [reply]
监控进程使用脚本编写即可，其中用 curl, wget 等工具监控是否可访问 google.com 即可，不要用 ping 等使用 icmp 协议的命令，因为代理协议（http(s)，socks等）不能代理 icmp 协议。

**Comment by sophauer** [comment]
后来我用reuse port 跟监控脚本实现了这个功能.确实如大佬所说的,并不需要依赖ss-tproxy来做
